spark/src/main/scala/org/apache/spark/sql/delta/DeltaHistoryManager.scala

```diff
@@ -115,8 +117,60 @@ class DeltaHistoryManager(
           }
         }.map(DeltaHistory.fromCommitInfo)
       }
-    // Spark should return the commits in increasing order as well
-    monotonizeCommitTimestamps(info.collect()).reverse
+    val monotonizedCommits = if (isRangeICT) {
+      info.collect()
+    } else {
+      monotonizeCommitTimestamps(info.collect())
+    }
+    monotonizedCommits.reverse
+  }
+
+  /**
+   * Get the commit information of the Delta table from commit `[start, end)`. If `end` is `None`,
+   * we return all commits from start to now.
+   * @param start The start of the commit range, inclusive.
+   * @param end The end of the commit range, exclusive.
+   */
+  def getHistory(
+      start: Long,
+      end: Option[Long] = None,
+      snapshotOpt: Option[Snapshot] = None): Seq[DeltaHistory] = {
```

Comment:
Remove `snapshotOpt` and rely on unsafeVolatileSnapshot instead. That will simplify the interface without any performance regressions.